services/distribution/src/main/java/app/coronawarn/server/services/distribution/statistics/local/LocalStatisticsToProtobufMapping.java

```diff
@@ -0,0 +1,228 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class LocalStatisticsToProtobufMapping {
+
+  private static final Logger logger = LoggerFactory.getLogger(LocalStatisticsToProtobufMapping.class);
+
+  private final StatisticJsonFileLoader jsonFileLoader;
+  private final LocalStatisticsDownloadService localStatisticsDownloadService;
+  private final RegionMappingConfig regionMappingConfig;
+
+  /**
+   * Process the JSON file provided by TSI and map the it to Local Statistics protobuf object.
+   *
+   * @param jsonFileLoader - Loader of the file from the system
+   * @param localStatisticsDownloadService - Local Statistics Download Service for keeping track of ETags
+   */
+  public LocalStatisticsToProtobufMapping(StatisticJsonFileLoader jsonFileLoader,
+      LocalStatisticsDownloadService localStatisticsDownloadService,
+      RegionMappingConfig regionMappingConfig) {
+    this.jsonFileLoader = jsonFileLoader;
+    this.localStatisticsDownloadService = localStatisticsDownloadService;
+    this.regionMappingConfig = regionMappingConfig;
+  }
+
+  /**
+   * Create protobuf local statistic map from raw JSON local statistics.
+   * Local statistics has to be uploaded on CDN in 7 packages. Each entry on the map represents 1 package.
+   * Packages of local statistics contains data about 1 up to 4 federal states and all their administrative units.
+   * For more info related to the grouping of the federal states into packages please look into
+   * https://github.com/corona-warn-app/cwa-app-tech-spec/blob/42e9e4f3c588cd2fd283f904e9f0ccd53a2b83d0/
+   * docs/spec/statistics.md#populate-local-statistical-data
+   *
+   * @return map containing local statistics grouped by archive id.
+   */
+  @Bean
+  public Map<Integer, LocalStatistics> constructProtobufLocalStatistics() {
+    Map<Integer, LocalStatistics> localStatisticsMap = new HashMap<>();
+
+    try {
+      Optional<JsonFile> optionalFile = this.getFile();
+
+      if (optionalFile.isEmpty()) {
+        logger.warn("Stats file is already updated to the latest version. Skipping generation.");
+        return Collections.emptyMap();
+      } else {
+        JsonFile file = optionalFile.get();
+        this.updateETag(optionalFile.get().getETag());
+
+        List<LocalStatisticsJsonStringObject> onePerProvinceStatistics = deserializeAndValidate(file);
+
+        onePerProvinceStatistics.forEach(localStatisticsJsonStringObject -> {
+          if (localStatisticsJsonStringObject.getProvinceCode() != null) {
+            int provinceCode = Integer.parseInt(localStatisticsJsonStringObject.getProvinceCode());
+
+            if (isFederalState(localStatisticsJsonStringObject.getProvinceCode())) {
```

Comment:
if you have parsed already the province code into `int`, why then not operating on the same?

`isFederalState(int i) { return i < 100; } `

Reply:
Yes, I will change it to this.